CHANGELOG.md

```diff
@@ -1,5 +1,15 @@
 # Change log
 
+## 1.1.0dev (unreleased)
+
+* `get-mathlib-cache` no longer understands `--rev`; if you want to use a
+  different mathlib version, edit your `leanproject.toml`.
+* Add `--fallback` to `get-cache` for traversing the git history to find an
+  approximate cache.
+* `get-cache` no longer modifies `lean` files in the working directory.
+* `mk-cache --force` no longer associates the cache with the current HEAD
```

Comment:
Personally, I think I would prefer if `leanproject` would categorically deny working on a dirty repo. It will probably save people quite some stress and tears.

Reply:
This happens to me all the time:
* Checkout a branch
* Start editing a file `foo.lean`, and save it
* Remember I didn't run `get-cache` yet

With this PR, `get-cache --force` will get the fresh oleans from azure, without causing the tears it currently causes by throwing away my changes to `foo.lean`.

Reply:
well, it creates a `git commit`, which might confuse users that are not very gitperienced.
What does `leanproject` do without the `--force`? Will it tell me that the directory is dirty, and that I should fix that or use `--force`? That would be good, I think.

Reply:
I personally like having the choice of doing `git stash` instead of `git commit`. So if `leanproject` doesn't do any git-stuff unless I `--force` it, then I'm happy.

Reply:
It doesn't really run either `git commit` or `git stash`; the commit it creates is invisible to any novice user of git, and only there if you're really looking for it.

At any rate, this comment chain might be more suited for #115

Reply:
Understood.

Reply:
```suggestion
* `mk-cache --force` no longer permits the working tree to be dirty.
```